### DISCUSSION TITLE
How to take a screenshot with avalonia

### DISCUSSION THREAD

**Discussion by sparrow2015** [AvaloniaUI/Avalonia#12779] [Q&A]
**How to take a screenshot with avalonia? I want to capture current full screen(not a control( e.g. button, text box etc ) screenshot,) and save to a png/jpg file**
Is there a demo or some tips?

**Comment by timunie** [comment]
Hi @cooper87, 

questions go into Discussions (Q&A) or join us on Telegram (https://t.me/Avalonia)

---

If you are taking about full screen capture, I guess this is a task for the OS and not for Avalonia. So you will need to write a service which you can inject per platform. For example windows: https://stackoverflow.com/a/39760420/13051114
